dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthorizationRestRepositoryIT.java

```diff
@@ -0,0 +1,1170 @@
+            .andExpect(status().isUnauthorized());
+
+        // verify that it works for normal loggedin users with an invalid or missing uri
+        getClient().perform(get("/api/authz/authorizations/search/object")
+                .param("uri", siteUri)
+                .param("eperson", eperson.getID().toString()))
+            .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    /**
+     * Verify that the findByObject return the 403 Forbidden response when a non-admin eperson try to search the
+     * authorization of another eperson
+     * 
+     * @throws Exception
+     */
+    public void findByObjectForbiddenTest() throws Exception {
+        Site site = siteService.findSite(context);
+        SiteRest siteRest = converterService.toRest(site, converterService.getProjection(DefaultProjection.NAME));
+        String siteUri = utils.linkToSingleResource(siteRest, "self").getHref();
+        context.turnOffAuthorisationSystem();
+        EPerson anotherEperson = EPersonBuilder.createEPerson(context).withEmail("another@example.com")
+                .withPassword(password).build();
+        context.restoreAuthSystemState();
+        // disarm the alwaysThrowExceptionFeature
+        configurationService.setProperty("org.dspace.app.rest.authorization.AlwaysThrowExceptionFeature.turnoff", true);
+        String anotherToken = getAuthToken(anotherEperson.getEmail(), password);
+        // verify that he cannot search the admin authorizations
+        getClient(anotherToken).perform(get("/api/authz/authorizations/search/object")
+                .param("uri", siteUri)
+                .param("eperson", admin.getID().toString()))
+            .andExpect(status().isForbidden());
+
+        // verify that he cannot search the authorizations of another "normal" eperson
+        getClient(anotherToken).perform(get("/api/authz/authorizations/search/object")
+                .param("uri", siteUri)
+                .param("eperson", eperson.getID().toString()))
+            .andExpect(status().isForbidden());
+    }
+
+    @Test
+    /**
+     * Verify that an exception in the feature check will be reported back
+     * @throws Exception
+     */
+    public void findByObjectInternalServerErrorTest() throws Exception {
+        Site site = siteService.findSite(context);
+        SiteRest siteRest = converterService.toRest(site, converterService.getProjection(DefaultProjection.NAME));
+        String siteUri = utils.linkToSingleResource(siteRest, "self").getHref();
+
+        // verify that it works for administrators
+        String adminToken = getAuthToken(admin.getEmail(), password);
+        getClient(adminToken).perform(get("/api/authz/authorizations/search/object")
+                .param("uri", siteUri)
+                // use a large page so that the alwaysThrowExceptionFeature is invoked
+                // this could become insufficient at some point
+                .param("size", "100")
```

Comment:
Wouldn't it be better to replace 100 with `authzFeatureService.findAll().size()` (same for the similar tests below)

Reply:
it doesn't really make a difference because we have an upper limit set by our pagination. I have preferred to avoid an extra call and also decided to set an explicit (reasonable limit) here with a commit so that if we hit it in future it should be easier to understand the issue. Using a larger value could stop to work when we will put in place a protection for our pagination to say "refuse request with page larger than 100, 500 or...".
When (if) we will hit this limit we will need to run across all the pages to be sure to hit the exception

Reply:
Do you expect the total amount of features to be > 100? If this is to be expected, it would be best to create a JIRA issue since we'd expect it to be a required modification

Reply:
https://jira.lyrasis.org/browse/DS-4463